core/profile.go

```diff
@@ -178,3 +187,118 @@ func (n *OpenBazaarNode) updateProfileCounts() error {
 }
+
+func ValidateProfile(profile *pb.Profile) error {
+	if len(profile.Handle) > WordMaxCharacters {
+		return fmt.Errorf("Handle character length is greater than the max of %d", WordMaxCharacters)
+	}
+	if len(profile.Name) > WordMaxCharacters {
+		return fmt.Errorf("Name character length is greater than the max of %d", WordMaxCharacters)
+	}
+	if len(profile.Location) > WordMaxCharacters {
+		return fmt.Errorf("Location character length is greater than the max of %d", WordMaxCharacters)
+	}
+	if len(profile.About) > AboutMaxCharacteres {
+		return fmt.Errorf("About character length is greater than the max of %d", AboutMaxCharacteres)
+	}
+	if len(profile.ShortDescription) > ShortDescriptionLength {
+		return fmt.Errorf("Short description character length is greater than the max of %d", ShortDescriptionLength)
+	}
+	if len(profile.Website) > URLMaxCharacters {
+		return fmt.Errorf("Website character length is greater than the max of %d", URLMaxCharacters)
+	}
+	if len(profile.Email) > SentenceMaxCharacters {
+		return fmt.Errorf("Email character length is greater than the max of %d", SentenceMaxCharacters)
+	}
+	if len(profile.PhoneNumber) > WordMaxCharacters {
+		return fmt.Errorf("Phone number character length is greater than the max of %d", WordMaxCharacters)
+	}
+	if len(profile.Social) > MaxListItems {
+		return fmt.Errorf("Number of social accounts is greater than the max of %d", MaxListItems)
+	}
+	for _, s := range profile.Social {
+		if len(s.Username) > WordMaxCharacters {
+			return fmt.Errorf("Social username character length is greater than the max of %d", WordMaxCharacters)
+		}
+		if len(s.Type) > WordMaxCharacters {
+			return fmt.Errorf("Social account type character length is greater than the max of %d", WordMaxCharacters)
+		}
+		if len(s.Proof) > URLMaxCharacters {
+			return fmt.Errorf("Social proof character length is greater than the max of %d", WordMaxCharacters)
+		}
+	}
+	if profile.ModInfo != nil {
+		if len(profile.ModInfo.Description) > AboutMaxCharacteres {
+			return fmt.Errorf("Moderator description character length is greater than the max of %d", AboutMaxCharacteres)
+		}
+		if len(profile.ModInfo.TermsAndConditions) > PolicyMaxCharacters {
+			return fmt.Errorf("Moderator terms and conditions character length is greater than the max of %d", PolicyMaxCharacters)
+		}
+		if len(profile.ModInfo.Languages) > MaxListItems {
+			return fmt.Errorf("Moderator number of languages greater than the max of %d", MaxListItems)
+		}
+		for _, l := range profile.ModInfo.Languages {
+			if len(l) > WordMaxCharacters {
+				return fmt.Errorf("Moderator language character length is greater than the max of %d", WordMaxCharacters)
+			}
+		}
+		if profile.ModInfo.Fee != nil {
+			if profile.ModInfo.Fee.FixedFee != nil {
+				if len(profile.ModInfo.Fee.FixedFee.CurrencyCode) > WordMaxCharacters {
+					return fmt.Errorf("Moderator fee currency code character length is greater than the max of %d", WordMaxCharacters)
```

Comment:
@jjeffryes What do you think about getting some back-end stories in the backlog so the server could maintain a valid list of countries and currency codes and then validate that any related fields give real values?

Otherwise, as it is now, someone could enter 'pickles' for a currency and 'marmalade' for a country.

It would be a slight pain to have to keep server and client lists in sync, but I think it would be less so than the client having to deal with bogus values.